Namespaces that start with a number cause errors

#### Description
In K8s and OCP you can create a namespace that starts with a number such as 246project, CC fails to create a VIP with this error:
2018/06/28 21:07:00 [INFO] Text: u'{"code":400,"message":"01070088:3: The requested object name (/k8s/246project_vipf5demo.poolonly) is invalid.","errorStack":[],"apiError":3}'
2018/06/28 21:07:00 [ERROR] [2018-06-28 21:07:00,256 f5_cccl.service.manager ERROR] Resource /k8s/246project_vipf5demo.poolonly creation error, requeuing task...

Pools are created successfully because they are prefaced with cfgmap_ or ingress_, we need the same functionality with VIP creation.

#### Kubernetes Version
Any
#### Controller Version
1.5

#### BIG-IP Version
13.x

#### Diagnostic Information
<Configuration files, error messages, logs>
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137

--- a/pkg/appmanager/appManager_test.go
+++ b/pkg/appmanager/appManager_test.go
@@ -4200,6 +4200,30 @@ var _ = Describe("AppManager Tests", func() {
 				Expect(err).ToNot(BeNil())
 			})
 
+			// This test case validates namespace name update funcitonality
+			// where, a name starting with a 'number' is prefixed with 'cfgmap_<name>'
+			It("Check namespace name starting with number", func() {
+				NumNSname := "4-default-rack"
+				cfgMapSelector, err := labels.Parse(DefaultConfigMapLabel)
+				Expect(err).To(BeNil())
+				err = mockMgr.appMgr.AddNamespace(NumNSname, cfgMapSelector, 0)
+				Expect(err).To(BeNil())
+
+				cfgFoo := test.NewConfigMap("foomap", "1", NumNSname,
+					map[string]string{
+						"schema": schemaUrl,
+						"data":   configmapFoo,
+					})
+
+				r := mockMgr.addConfigMap(cfgFoo)
+				Expect(r).To(BeTrue(), "Config map should be processed.")
+				resources := mockMgr.resources()
+				rs, ok := resources.Get(serviceKey{"foo", 80, NumNSname},
+					formatConfigMapVSName(cfgFoo))
+				Expect(ok).To(BeTrue(), "Config map should be accessible.")
+				Expect(rs.Virtual.Name).To(Equal("cfgmap_" + NumNSname + "_foomap"))
+			})
+
 			It("watches namespace labels", func() {
 				nsLabel := "watching"
 				err := mockMgr.startLabelMode(nsLabel)

--- a/pkg/appmanager/resourceConfig.go
+++ b/pkg/appmanager/resourceConfig.go
@@ -222,8 +222,16 @@ func (v *Virtual) SetVirtualAddress(bindAddr string, port int32) {
 }
 
 // format the virtual server name for a ConfigMap
+
+// To handle VS name which start with Number,
+// we are prefixing with 'cfgmap_' to avoid errors with bigip.
 func formatConfigMapVSName(cm *v1.ConfigMap) string {
-	return fmt.Sprintf("%s_%s", cm.ObjectMeta.Namespace, cm.ObjectMeta.Name)
+	VSprefix := "cfgmap"
+	if _, err := strconv.Atoi(cm.ObjectMeta.Namespace[0:1]); err == nil {
+		return fmt.Sprintf("%s_%s_%s", VSprefix, cm.ObjectMeta.Namespace, cm.ObjectMeta.Name)
+	} else {
+		return fmt.Sprintf("%s_%s", cm.ObjectMeta.Namespace, cm.ObjectMeta.Name)
+	}
 }
 
 // format the pool name for a ConfigMap

